lib/puma/sd_notify.rb

```diff
@@ -0,0 +1,124 @@
+# frozen_string_literal: true
+
+require "socket"
+
+# SdNotify is a pure-Ruby implementation of sd_notify(3). It can be used to
```

Comment:
What's the provenance of this code/where'd it come from? Did you write this?

Reply:
Looks like it comes from https://github.com/mperham/sidekiq/pull/4489/files#diff-aeecdbde3c294612b90dbc578e369375

Reply:
Ah I see from your description now.

I think we will have to include the original LICENSE at the top of the file. https://github.com/agis/ruby-sdnotify/blob/master/LICENSE

Reply:
I'm also interested in discussing if we should simply use the gem, and raise/bail out if someone tries to use the feature without the correct constant defined.

Reply:
Sorry, I thought that I put the license it was one of my concerns, I will fix that.

Reply:
I agree with you to use the gem. Also, the quality of gem's code is very good and the sysmtemd specification is stable so we will not do other changes on the gem's code in the near feature IMHO.